mlflow/models/evaluation/default_evaluator.py

```diff
@@ -369,6 +380,31 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _compute_df_mode_or_mean(df):
+    """
+    Compute mode (for integer or non-numeric columns or mean (for float columns) for the
+    input dataframe, return a dict, key is column name, value is the corresponding mode or
+    mean value.
+    """
+    # convert pandas dataframe columns to best possible dtypes.
+    # e.g. if one column has float type values, but they are actually all integers,
+    # it will be converted to integer column
+    categorical_cols = []
+    continuous_cols = []
+    for col in df.columns:
+        if _is_categorical_values(df[col].tolist()):
```

Comment:
```suggestion
        if _is_categorical_values(df[col]):
```

Do we need `tolist()`?